### ISSUE TITLE
[Accessibility] Narrator should hide to announce the text in textBox/maskedTextBox when enable UseSystemPasswordChar or PasswordChar properties

### ISSUE THREAD

**Issue by Olina-Zhang** [dotnet/winforms#7852]
### .NET version

**.Net 8.0 SDK build:** 8.0.100-alpha.1.22476.24

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No, not a regression issue

### Issue description

When typing text in TextBox/MaskedTextBox with UseSystemPasswordChar or PasswordChar, Narrator announce text to "Hidden", but when focus on TextBox/MaskedTextBox or specific text, it will announce as specific text. 

![image](https://user-images.githubusercontent.com/26474449/192480453-0a1cf219-1500-4a4d-aa8f-911e473736b6.png)

### Steps to reproduce

Use the sample application to test: 
[TextBoxPasswordSample.zip](https://github.com/dotnet/winforms/files/9653779/TextBoxPasswordSample.zip)

**Comment by v-elnovikova** [comment]
PR #7741 added hiding `IValueProvider.Value` value for password-protected textboxes. Although the issue still reproduces occasionally. Also, when navigating through textbox with arrow keys, Narrator can announce password chars one-by-one:

![Narrator-leaks-password](https://github.com/dotnet/winforms/assets/113603457/2637a45d-a5c7-4d5b-a9be-1fc20702ad10)

https://github.com/dotnet/winforms/assets/113603457/7751bfa1-c9aa-4a27-951a-68314d2054bb

**Comment by Olina-Zhang** [comment]
Verified this issue in main branch for Winforms repo, it was fixed: Now Narrator cannot announce the the plain text for textBox/maskedTextBox with UseSystemPasswordChar or PasswordChar setting:

Narrator:
![image](https://github.com/dotnet/winforms/assets/26474449/61de6596-aba2-46ac-9f87-a036002d2df0)
Inspect:
![image](https://github.com/dotnet/winforms/assets/26474449/e1dcf7e5-ff88-45cd-b731-31c7723f5e52)
